impl/proc_macros/src/typed_array_element.rs

```diff
@@ -0,0 +1,393 @@
+use proc_macro2::{Span, TokenStream};
+use syn::parse::{Parse, ParseStream};
+use syn::{Ident, Type};
+
+#[derive(Debug)]
+struct Args {
+    trait_name: Ident,
+    rust_ty: Type,
+    sys_ty: Option<Type>,
+    sys_ref_ty: Option<Type>,
+    infix: String,
+}
+
+impl Parse for Args {
+    fn parse(input: ParseStream) -> Result<Self, syn::Error> {
+        input.parse::<Token![impl]>()?;
+
+        let trait_name = input.parse()?;
+
+        input.parse::<Token![for]>()?;
+
+        let rust_ty = input.parse()?;
+
+        let sys_ty = input
+            .parse::<Token![as]>()
+            .ok()
+            .map(|_| input.parse())
+            .transpose()?;
+
+        let sys_ref_ty = input
+            .parse::<Token![ref]>()
+            .ok()
+            .map(|_| input.parse())
+            .transpose()?;
+
+        input.parse::<Token![=>]>()?;
+
+        let infix = input.parse::<Ident>()?;
+        let infix = infix.to_string();
+
+        let brace_content;
+        braced!(brace_content in input);
+
+        brace_content.parse::<Token![..]>()?;
+
+        if !brace_content.is_empty() {
+            return Err(brace_content.error("expecting only `..` in braces"));
+        }
+
+        if !input.is_empty() {
+            return Err(input.error("expecting end of macro input"));
+        }
+
+        Ok(Args {
+            trait_name,
+            rust_ty,
+            sys_ty,
+            sys_ref_ty,
+            infix,
+        })
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+struct Env<'a> {
+    infix: &'a str,
+
+    rust_ty: &'a Type,
+    sys_ty: &'a Type,
+    sys_ref_ty: &'a Type,
+
+    sys_array: &'a Type,
+    sys_read_access: &'a Type,
+    sys_write_access: &'a Type,
+}
+
+static METHODS: &[&str] = &[
+    "new",
+    "new_copy",
+    "new_with_array",
+    "append",
+    "append_array",
+    "insert",
+    "invert",
+    "push_back",
+    "remove",
+    "resize",
+    "read",
+    "write",
+    "set",
+    "get",
+    "size",
+    "destroy",
+    "read_access_copy",
+    "read_access_ptr",
+    "read_access_operator_assign",
+    "read_access_destroy",
+    "write_access_copy",
+    "write_access_ptr",
+    "write_access_operator_assign",
+    "write_access_destroy",
+];
+
+fn impl_fn_symbol(method: &str) -> Ident {
+    Ident::new(&format!("{}_fn", method), Span::call_site())
+}
+
+fn fn_symbol(infix: &str, method: &str) -> Ident {
+    Ident::new(
+        &format!("godot_pool_{}_array_{}", infix, method),
```

Comment:
```suggestion
fn fn_symbol(type_: &str, method: &str) -> Ident {
    Ident::new(
        &format!("godot_pool_{}_array_{}", type_, method),
```

Reply:
Changed all occurrences of `infix` to `godot_type` just to be a tiny bit clearer.